### PR TITLE
Auto-complete for profile property rules

### PR DESCRIPTION
**Pull request by andyjih**
Screen Shot 2020-04-23 at 2 36 04 PM

https://www.pivotaltracker.com/story/show/172181082

### CODE REVIEW

File: core/web/components/forms/profilePropertyRule/edit.tsx [hunk 1]
```diff
@@ -37,6 +58,7 @@ export default function ({
     },
   });
   const [localFilters, setLocalFilters] = useState([]);
+  // const [selected, setSelected] = useState([]);
```
Comment (evantahler) [inline core/web/components/forms/profilePropertyRule/edit.tsx]: remove please

File: core/web/components/forms/profilePropertyRule/edit.tsx [hunk 2]
```diff
@@ -281,6 +303,49 @@ export default function ({
...
+                        </small>,
+                      ];
+                    }}
+                    defaultSelected={
+                      profilePropertyRule.options[opt?.key]
+                        ? [profilePropertyRule.options[opt?.key]]
+                        : undefined
+                    }
+
+                    // defaultSelected={[profilePropertyRule.options[opt?.key]]}
```
Comment (evantahler) [inline core/web/components/forms/profilePropertyRule/edit.tsx]: remove please

File: core/web/components/forms/profilePropertyRule/edit.tsx [hunk 3]
```diff
@@ -1,11 +1,20 @@
...
 import { useApi } from "../../../hooks/useApi";
-import { Row, Col, Button, Form, Table, Badge } from "react-bootstrap";
+import {
+  Row,
+  Col,
+  Button,
+  Form,
+  Table,
+  Badge,
+  Dropdown,
```
Comment (evantahler) [inline core/web/components/forms/profilePropertyRule/edit.tsx]: I don't think this is being used?

File: plugins/@grouparoo/csv/src/lib/file-import/profilePropertyRuleOptions.ts [hunk 4]
```diff
@@ -6,7 +6,7 @@ export const profilePropertyRuleOptions: PluginConnectionProfilePropertyRuleOpti
     key: "column",
     required: true,
     description: "where the data comes from",
-    type: "list",
+    type: "typeahead",
```
Comment (evantahler) [inline plugins/@grouparoo/csv/src/lib/file-import/profilePropertyRuleOptions.ts:9]: woo! new type!

**Review by evantahler — CHANGES_REQUESTED** [review]

File: core/web/components/forms/profilePropertyRule/edit.tsx [hunk 5]
```diff
@@ -281,6 +303,49 @@ export default function ({
...
+                              ? opt.examples.slice(0, 3).join("").trim() !== ""
+                                ? opt.examples.slice(0, 3).join(", ")
+                                : "None"
+                              : null}
+                          </em>
+                        </small>,
+                      ];
+                    }}
+                    defaultSelected={
+                      profilePropertyRule.options[opt?.key]
```
Comment (bleonard) [inline core/web/components/forms/profilePropertyRule/edit.tsx:331]: can be simplified as just `profilePropertyRule.options[opt?.key]` probably beccause that returns null or undefined itself.

**Comment by bleonard** [comment]
Looks nice!

File: core/web/components/forms/profilePropertyRule/edit.tsx (hunk 5 above)
Comment (andyjih) [reply core/web/components/forms/profilePropertyRule/edit.tsx:331]: I tried that at first, but it freaked out when it was null/undefined.

Comment (evantahler) [reply core/web/components/forms/profilePropertyRule/edit.tsx:331]: note that there's an array type or undefined (vs null).  This lib cares about the distinction!

**Review by evantahler — APPROVED** [review]

File: core/web/components/forms/profilePropertyRule/edit.tsx (hunk 5 above)
Comment (bleonard) [reply core/web/components/forms/profilePropertyRule/edit.tsx:331]: So maybe `profilePropertyRule.options[opt?.key] || undefined` then

**Pull request merged**